### PR TITLE
Add deactivateProduct() API call

### PR DESCRIPTION
**Pull request by skazi0**
This is used in both registration and deregistration.
Additional Product field was added to match original implementation.

Some behavior differences wrt original implementation were noted in
DIFFERENCES.md

### CODE REVIEW

File: connect/api_test.go [hunk 1]
```diff
@@ -166,3 +166,44 @@ func TestUpgradeProductError(t *testing.T) {
 		}
 	}
 }
+
+func TestDeactivateProductSMT(t *testing.T) {
+	createTestCredentials("", "", t)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(readTestFile("service_inactive_smt.json", t))
```
Comment (djoreilly) [inline connect/api_test.go:174]: this file is missing

**Review by djoreilly — CHANGES_REQUESTED** [review]

File: connect/api_test.go (hunk 1 above)
Comment (skazi0) [reply connect/api_test.go:174]: added

**Comment by skazi0** [comment]
I added another test sample for SCC vs SMT comparison.

**Review by djoreilly — APPROVED** [review]
lgtm

**Pull request merged**